pkg/client/queue/queue.go

```diff
@@ -0,0 +1,103 @@
+package queue
```

Comment:
Why do we need a second queue type in addition to the one in the `api` package? Adding new types, files, and code carries a significant cost in terms of system complexity and ease of understanding.

Reply:
Here is my reasoning process:
1. api package is auto generated gRPC package. It defines request and response types used in gRPC protocol. It also doesn't feel right of adding methods to autogenerated types. Armada server business logic (repository abstraction for example) is coupled to autogenerated types. This is an odd design by itself, as API directly exposes and is coupled to public available data structures.  Any changes to API indirectly changes the repo layer and vice-versa.
2. Introducing new package decouples public api from business and repo layers.
3. Any business logic should depend on this `queue` package as it defines a valid source of truth for Queue data. If used properly, any queue from this package can be considered valid (valid values for priority factor, resource limits etc...)
4. The changes in this pull request redefine how queue data structure looks like. User owners and  Group owners have been removed and have been replaced with permissions, subjects and verbs. The new type ensures transformation between two version of Queue. These transformation are required for gRPC request and from reading data from repositories.